spec/controllers/external_authors_controller_spec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe ExternalAuthorsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { FactoryGirl.create(:user) }
+  let(:invitation) { FactoryGirl.create(:invitation, external_author: external_author) }
+  let(:external_author) { FactoryGirl.create(:external_author) }
+
+  before(:each) do
+    fake_login_known_user(user)
+  end
+
+  describe "GET #get_external_author_from_invitation" do
+    context "with invalid invitation token" do
+      it "redirects with an error" do
+        get :get_external_author_from_invitation, invitation_token: "None existent"
+        it_redirects_to_with_error(root_path, "You need an invitation to do that.")
+      end
+    end
+
+    context "with valid invitation token" do
+      xit "assigns invitation" do
+        # ActionView::MissingTemplate:
+        # Missing template external_authors/get_external_author_from_invitation, application/get_external_author_from_invitation with {:locale=>[:en], :formats=>[:html], :handlers=>[:erb, :builder]}. Searched in:
+        get :get_external_author_from_invitation, invitation_token: invitation.token
+        expect(assigns(:invitation)).to eq(invitation)
+        assert_equal response.status, 200
+      end
+    end
+
+    context "without works to claim" do
+      it "redirects with an error" do
+        no_story_invitation = FactoryGirl.create(:invitation)
+        get :get_external_author_from_invitation, invitation_token: no_story_invitation.token
+        it_redirects_to_with_error(signup_path(no_story_invitation.token), "There are no stories to claim on this invitation. Did you want to sign up instead?")
+      end
+    end
+  end
+
+  describe "GET #complete_claim" do
+    it "redirects with a success message" do
+      get :complete_claim, invitation_token: invitation.token
+      it_redirects_to_with_notice(user_external_authors_path(user), "We have added the stories imported under #{external_author.email} to your account.")
```

Comment:
@annelu This doesn't work for me.
  describe "GET #complete_claim" do
    it "redirects with a success message" do
      expect(invitation).to receive(:mark_as_redeemed)
      expect(external_author).to receive(:claim!)
      get :complete_claim, invitation_token: invitation.token
      it_redirects_to_with_notice(user_external_authors_path(user), "We have added the stories imported under #{external_author.email} to your account.")
    end
  end